### ISSUE TITLE
Salt PPM Levels

### ISSUE THREAD

**Issue by Kranzy** [sfeakes/AquapureD#3]
So just thinking about the PPM levels for salt. I had a water test done by the pool shop today and my salt level is really high, about 8000 ppm. 

Previously Aquapure was showing readings of 4000, earlier today it dropped down to 2900 (not sure why)

The readings that come from the SWG, would they need to be calibrated to suit the levels of the chlorinator? Manual for my chlorinator states ideal Salt PPM Range: 4000 - 7000.

**Comment by sfeakes** [comment]
If you have a good SWG reading, you can calibrate many SWG to suit that calibration. (Have to check the manual for that).  I personally don't trust pool shops in my area as they never give me good readings, always on the low side so they can try to sell me more crap.
But after adding salt, many SWG will take at least 2 days to read the correct value.

**Comment by Kranzy** [comment]
Manual doesn't state anything about calibrating PPM levels. 

So the values that AquapureD gets from the SWG is the actual PPM reading and not a percentage of the range of that device or equivalent? 

Just checking that our Australian units which seem to be setup for much higher PPM values 4000-7000 range will work without code changes?

**Comment by sfeakes** [comment]
PPM is taken from the RS485 message the SWG sends back to AquapureD. It shouldn’t require any code changes even with the PPM values used in Australia.  The max value the PPM bit can hold is 25,400.  So I wouldn’t think the Tri units use a different formula as there is no need.
Have you seen the PPM change since adding salt, as said before it will take a few days to reflect.

**Comment by Kranzy** [comment]
I havnt added salt. No need to as my reading was high (8000 - pool shop result). Tri was showing 4000 but is now showing 2900, not sure why. Pool shop test was done after the reading dropped to 2900.

**Comment by tortfeaser** [comment]
I'm not sure I'm getting correct readings either (Zodiac Tri pH), AquapureD shows 4000 constantly, despite putting in three bags a couple of weeks back. I get behaviour like @Kranzy.

My guess is that when salt is measured as below 4000 it will show the value, when greater than 4000 it shows 4000. The 4000 point is the point at which the low salt warning light is displayed.

I'm keen to install an EC sensor to compare the Zodiac value to another measurement over time. Maybe I'll do it sooner rather than later.

**Comment by Kranzy** [comment]
Mine is currently reading at 2900 but no low salt light on the SWG unit, which I wouldnt expect as the measured value from the pool shop was up at nearly 8000.

**Comment by sfeakes** [comment]
The measurement works accurately for Haywood / Goldline / Pentair & Jandy SWG, so I'm really surprised that Zodiac would use something different.  This is how it's calculated.
```Received   AR PPM | HEX: 0x10|0x02|0x00|0x16|0x28|0x00|0x00|0x00|0x50|0x10|0x03|```
The 5 bit is the PPM ( 0x28 ).  0x28 hex is 40 in dec, then simply multiply by 100 for PPM. = 4000
That bit can go to 0xFF = 255 in decimal, then * 100 would hit 255,000, so you can get to the PPM levels used in Australia without needing to change the protocol.

If you run in debug serial mode you will see these messages.  Maybe it's worth monitoring them and your true PPM levels and see if you think we should change the calculation for the Zri units.

**Comment by Kranzy** [comment]
Yeah I'm wondering if its not updating correctly because if that was the true reading from the SWG I would expect to get a low salt warning, so not sure what is going on.

**Comment by Kranzy** [comment]
Hi @sfeakes,

Just to add to this, since my ppm dropped to 2900 it has never changed. Even after haivng to throw in 6 bags of salt due to excess rain we've had which had drastically dropped then increased my salt reading I dont see a change in the ppm reading coming from aquapured.

**Comment by Kranzy** [comment]
@sfeakes is there a way to reset the ppm value as I have not seen a change since it sitting on 2900, unless there is an issue with what the cell is reporting to the SWG, althgouh I would expect a low salt warning at that number.

**Comment by Kranzy** [comment]
Just to update this, I was not getting any notification on AquapureD regarding 'check cell' so I was going to leave it for future diagnostics but I decided to reset the cell counter today and my PPM value jumped up to 4000. 

Just went to the pool shop and did a test and it was at 4600. I need to put a couple of bags of salt in so I will see if the readings change at all once that has been done.
